### ISSUE TITLE
Hamburger menu adds extra margin on left side

### ISSUE THREAD

**Issue by pjebs** [carbon-design-system/carbon-components-svelte#1145]
I added a hamburger menu and it's created an extra margin on the left side. My UI now looks weird because the Rows>Tiles (that populate the Content) have a larger margin on the left side compared to the right side (even when menu is not open). (i.e. the margins are unequal)

Is this intended behaviour? If so, then shouldn't it be changed.

<img width="303" alt="Screen Shot 2022-03-05 at 12 16 15 pm" src="https://user-images.githubusercontent.com/5377970/156861711-7ced0e67-7b50-4053-b9cd-b5f172ba2213.png">

<img width="246" alt="Screen Shot 2022-03-05 at 12 16 35 pm" src="https://user-images.githubusercontent.com/5377970/156861714-52c484c6-7ca6-423f-b50d-9b4cdaabedf1.png">

**Comment by metonym** [comment]
Could you provide a minimal repro? That would greatly aid in debugging/investigating this issue.

**Comment by AlvaroPata** [comment]
I have a similar issue. When the left side nav is closed, the content gets an extra margin of 3rem on the left.
<img width="1302" alt="Screenshot 2022-03-14 at 15 55 55" src="https://user-images.githubusercontent.com/69887666/158219799-3d155789-f278-496f-9354-6a27e48228bd.png">
<img width="1302" alt="Screenshot 2022-03-14 at 15 56 17" src="https://user-images.githubusercontent.com/69887666/158219775-2bba2915-ff75-49ad-a259-eaa8057063bd.png">

I barely changed the code that I copied. This is what my layout page looks like
```svelte
<script>
    import {
        Header,
        HeaderUtilities,
        HeaderAction,
        HeaderPanelLinks,
        HeaderPanelLink,
        SideNav,
        SideNavItems,
        SideNavDivider,
        SideNavMenu,
        SideNavMenuItem,
        SideNavLink,
        SkipToContent,
        Content,
    } from 'carbon-components-svelte';
    import { expoIn } from 'svelte/easing';

    let isSideNavOpen = false;
    let isOpen = false;
</script>

<Header platformName="archpy" expansionBreakpoint={1312} bind:isSideNavOpen>
    <svelte:fragment slot="skip-to-content">
        <SkipToContent />
    </svelte:fragment>
    <HeaderUtilities>
        <HeaderAction bind:isOpen transition={expoIn}>
            <HeaderPanelLinks>
                <HeaderPanelLink href="/login">Login</HeaderPanelLink>
                <HeaderPanelLink>Admin</HeaderPanelLink>
                <HeaderPanelLink>API</HeaderPanelLink>
            </HeaderPanelLinks>
        </HeaderAction>
    </HeaderUtilities>
</Header>

<SideNav bind:isOpen={isSideNavOpen}>
    <SideNavItems>
        <SideNavLink text="Home" href="/" />
        <SideNavLink text="About us" href="/about" />
        <SideNavMenu text="Competitions">
            <SideNavMenuItem href="/" text="Competition #1" />
        </SideNavMenu>
        <SideNavDivider />
        <SideNavLink text="Dashboard" href="/dashboard" />
    </SideNavItems>
</SideNav>

<Content>
    <slot />
</Content>

```

**Comment by AlvaroPata** [comment]
The `Content` component also adds a background colour that is especially noticeable with darker themes.

<img width="1392" alt="Screenshot 2022-03-14 at 21 04 37" src="https://user-images.githubusercontent.com/69887666/158252371-1d457f73-2576-4899-ac74-4e9291e7575a.png">

After disabling these rules in the browser, it looked good again:
```css
.bx--side-nav ~ .bx--content {
	/* margin-left: 3rem; */
}

.bx--content {
	padding: 2rem;
	/* background: var(--cds-ui-02, #474d38); */
	will-change: margin-left;
}
```
<img width="1392" alt="Screenshot 2022-03-14 at 21 15 30" src="https://user-images.githubusercontent.com/69887666/158253827-8f35bfa4-26c1-4e12-9052-313514baf691.png">
Still, not quite sure how to fix it.

**Comment by metonym** [comment]
@AlvaroPata Upstream issue regarding the `.bx--content` background color: https://github.com/carbon-design-system/carbon/issues/9877

**Comment by endigma** [comment]
this seems fixed in upstream, how to make this work here?

**Comment by metonym** [comment]
> this seems fixed in upstream, how to make this work here?

Are you referring to the .bx-content background color?

**Comment by endigma** [comment]
https://github.com/carbon-design-system/carbon/pull/10992 is merged

**Comment by metonym** [comment]
@endigma That was released as part of `carbon-components` version 10.56.0, which is now supported in [v0.63.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.63.0).

**Comment by iohzrd** [comment]
I'm experiencing this issue as well. Putting the side nav elements inside the header "fixes" the margin issue, but then the side nav floats on top of the content instead of pushing it to the side.

**Comment by willbicks** [comment]
Having the same issue as @iohzrd, I would love to be able to have the side navs occupy space on the left hand side on desktop, which requires that the `SideNav` be outside my `Header`. However, this means that when the menu collapses into the hamburger on mobile, the left margin on `<main id="main-content" class="bx--content">` takes up unnecessary space.

**Comment by willbicks** [comment]
Interestingly, the demos (eg: https://carbon-components-svelte.onrender.com/framed/UIShell/HeaderNav?theme=g100) keep the SideNav outside the header, and yet don't have this issue. Below is a side by side of the exact same `.svelte` file as listed in the docs site running in my local dev environment. I've tracked it down to the `@media` css query which is shown in the left debug window, but which is not generated by my environment. From inspection (and toggling it on and off) that rule is what appropriately removes the left margin when the side nav is collapsed.

![Side by side of margin issue on demo site vs local environment](https://user-images.githubusercontent.com/6686319/179421929-b54cfd2b-b1d2-43e6-88c9-f53b2fe3337f.png)

I have verified that I'm using the latest release, and this is a brand new fresh SvelteKit project. I haven't yet enabled `carbon-preprocess-svelte`.

**Comment by metonym** [comment]
The docs site overrides these styles: https://github.com/carbon-design-system/carbon-components-svelte/blob/f8e400f2edd971f41dacfbc21bfff4ee059e5163/docs/src/global.css#L247-L258

You can apply similar overrides to achieve the desired effect.

**Comment by willbicks** [comment]
@metonym, thanks for the tip! Given your suggestion, I wrote some Svelte logic to hack this using the `<Breakpoint />` component instead of hard-coding the pixel value. If anyone else wants this behavior I've included it below.

```svelte
<script lang="ts">
    import {
        Breakpoint
    } from "carbon-components-svelte";

    import type { BreakpointSize } from "carbon-components-svelte/types/Breakpoint/Breakpoint.svelte";

    let breakpoint: BreakpointSize;
</script>

<Breakpoint bind:size={breakpoint}/>

/* -- Navigation Components Here -- */

<div style="display: none;" class:nav-collapsed={breakpoint == "sm"}></div>

<style global>
/* Fix side margin when nav collapsed (see https://github.com/carbon-design-system/carbon-components-svelte/issues/1145) */
.bx--side-nav~.:nav-collapsed~.bx--content {
    margin-left: 0;
}
</style>
```

**Comment by willbicks** [comment]
On further investigation, this is a much simpler and less hacky way of achieving the same goal while not relying on a hardcoded pixel breakpoint. 

```svelte
<style global>
/* Fix side margin when nav collapsed (see https://github.com/carbon-design-system/carbon-components-svelte/issues/1145#issuecomment-1186623855) */
.bx--side-nav--collapsed~.bx--content {
    margin-left: 0;
}
</style>
```

Is there any reason this shouldn't be the default behavior? If not, I'd be happy to open a PR for this issue.

**Comment by metonym** [comment]
That's reasonable – PRs welcome.

**Comment by metonym** [comment]
Fixed in [v0.67.7](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.67.7).

If the `SideNav` is collapsed, `Content` will have a left margin of `0`. Furthermore, global styles in the docs site have been updated to avoid overriding UI Shell styles to avoid confusion.
